corvo-batcher/src/corvo_batcher.cc

```diff
@@ -0,0 +1,144 @@
+// Custom batcher for corvo, which:
+// - Prevents merging of batches.
+// - Limits the number of concurrent requests.
+
+#include <memory>
+#include <mutex>
+
+#include "triton/core/tritonbackend.h"
+
+namespace {
+
+constexpr uint64_t kNumConcurrentRequest = 16;
+
+// Semaphore implements a basic semaphore in C++, similar to
+// std::counting_semaphore in C++20.
+class Semaphore {
```

Comment:
Have we considered other libraries w/ semaphore i.e. boost? This seems fine for now given that it's simple